ballerina/ebcdic_ascii_transformer.bal

```diff
@@ -0,0 +1,561 @@
+    {ebcdic: 168, ascii: 121},
+    {ebcdic: 169, ascii: 122},
+    {ebcdic: 170, ascii: 26},
+    {ebcdic: 171, ascii: 26},
+    {ebcdic: 172, ascii: 26},
+    {ebcdic: 173, ascii: 26},
+    {ebcdic: 174, ascii: 26},
+    {ebcdic: 175, ascii: 26},
+    {ebcdic: 176, ascii: 26},
+    {ebcdic: 177, ascii: 26},
+    {ebcdic: 178, ascii: 26},
+    {ebcdic: 179, ascii: 26},
+    {ebcdic: 180, ascii: 26},
+    {ebcdic: 181, ascii: 26},
+    {ebcdic: 182, ascii: 26},
+    {ebcdic: 183, ascii: 26},
+    {ebcdic: 184, ascii: 26},
+    {ebcdic: 185, ascii: 26},
+    {ebcdic: 186, ascii: 26},
+    {ebcdic: 187, ascii: 26},
+    {ebcdic: 188, ascii: 26},
+    {ebcdic: 189, ascii: 26},
+    {ebcdic: 190, ascii: 26},
+    {ebcdic: 191, ascii: 26},
+    {ebcdic: 192, ascii: 123},
+    {ebcdic: 193, ascii: 65},
+    {ebcdic: 194, ascii: 66},
+    {ebcdic: 195, ascii: 67},
+    {ebcdic: 196, ascii: 68},
+    {ebcdic: 197, ascii: 69},
+    {ebcdic: 198, ascii: 70},
+    {ebcdic: 199, ascii: 71},
+    {ebcdic: 200, ascii: 72},
+    {ebcdic: 201, ascii: 73},
+    {ebcdic: 202, ascii: 26},
+    {ebcdic: 203, ascii: 26},
+    {ebcdic: 204, ascii: 26},
+    {ebcdic: 205, ascii: 26},
+    {ebcdic: 206, ascii: 26},
+    {ebcdic: 207, ascii: 26},
+    {ebcdic: 208, ascii: 125},
+    {ebcdic: 209, ascii: 74},
+    {ebcdic: 210, ascii: 75},
+    {ebcdic: 211, ascii: 76},
+    {ebcdic: 212, ascii: 77},
+    {ebcdic: 213, ascii: 78},
+    {ebcdic: 214, ascii: 79},
+    {ebcdic: 215, ascii: 80},
+    {ebcdic: 216, ascii: 81},
+    {ebcdic: 217, ascii: 82},
+    {ebcdic: 218, ascii: 26},
+    {ebcdic: 219, ascii: 26},
+    {ebcdic: 220, ascii: 26},
+    {ebcdic: 221, ascii: 26},
+    {ebcdic: 222, ascii: 26},
+    {ebcdic: 223, ascii: 26},
+    {ebcdic: 224, ascii: 92},
+    {ebcdic: 225, ascii: 26},
+    {ebcdic: 226, ascii: 83},
+    {ebcdic: 227, ascii: 84},
+    {ebcdic: 228, ascii: 85},
+    {ebcdic: 229, ascii: 86},
+    {ebcdic: 230, ascii: 87},
+    {ebcdic: 231, ascii: 88},
+    {ebcdic: 232, ascii: 89},
+    {ebcdic: 233, ascii: 90},
+    {ebcdic: 234, ascii: 26},
+    {ebcdic: 235, ascii: 26},
+    {ebcdic: 236, ascii: 26},
+    {ebcdic: 237, ascii: 26},
+    {ebcdic: 238, ascii: 26},
+    {ebcdic: 239, ascii: 26},
+    {ebcdic: 240, ascii: 48},
+    {ebcdic: 241, ascii: 49},
+    {ebcdic: 242, ascii: 50},
+    {ebcdic: 243, ascii: 51},
+    {ebcdic: 244, ascii: 52},
+    {ebcdic: 245, ascii: 53},
+    {ebcdic: 246, ascii: 54},
+    {ebcdic: 247, ascii: 55},
+    {ebcdic: 248, ascii: 56},
+    {ebcdic: 249, ascii: 57},
+    {ebcdic: 250, ascii: 26},
+    {ebcdic: 251, ascii: 26},
+    {ebcdic: 252, ascii: 26},
+    {ebcdic: 253, ascii: 26},
+    {ebcdic: 254, ascii: 26},
+    {ebcdic: 255, ascii: 26}
+];
+
+# Converts a byte array encoded in EBCDIC to a byte array encoded in ASCII.
+#
+# + ebcdic - The input byte array encoded in EBCDIC
+# + return - The output byte array encoded in ASCII
+public isolated function toAsciiBytes(byte[] ebcdic) returns byte[] {
```

Comment:
What if user provides an invalid encoded byte array ? Lets say user provides `ascii` encoded byte array here, what is the behavior ?

Reply:
it's users responsibility to pass the the correct encoded byte array. We don't have any mechanism to figure out the encoding as it's just bytes.